src/Item/Item.router.ts

```diff
@@ -394,37 +394,32 @@ export class ItemRouter extends Router {
           STATUS_CODES.unauthorized
         )
       }
-
-      const service = new CollectionService()
-
-      if (service.isLockActive(dbCollection.lock)) {
-        throw new HTTPError(
-          "Locked collection items can't be updated",
-          { id },
-          STATUS_CODES.locked
-        )
-      }
     }
 
     const isDbCollectionPublished = await getIsCollectionPublished(dbCollection)
+    const service = new CollectionService()
 
-    if (isDbCollectionPublished) {
-      if (!dbItem) {
-        throw new HTTPError(
-          "Items can't be added to a published collection",
-          { id },
-          STATUS_CODES.badRequest
-        )
-      }
+    if (isDbCollectionPublished && !dbItem) {
+      throw new HTTPError(
+        "Items can't be added to a published collection",
+        { id },
+        STATUS_CODES.badRequest
+      )
+    }
+
+    const isCollectionPublishedOrLocked =
+      isDbCollectionPublished ||
+      (dbCollection && service.isLockActive(dbCollection.lock))
 
+    if (dbItem && isCollectionPublishedOrLocked) {
```

Comment:
Can you add a test verifying that the rarity can't be changed if the collection is locked?